lab-integration/src/main.ts

```diff
@@ -1,47 +1,80 @@
-import { DateTime } from "luxon";
 import { promises as fs, watch } from "fs";
+import * as lockfile from "proper-lockfile";
 import path from "path";
-import { TestsResult } from "./types";
+import { queue } from "async";
```

Comment:
Have you considered the `fastq` library? It claims to be faster than `async`. Also, it's a lot smaller.

https://www.npmjs.com/package/fastq

Reply:
Acknowledge. By the way, I think this is better too 'cause is more specific and on that way fit our use case that is just queue task.